src/internal/ppsdb/ppsdb.go

```diff
@@ -48,12 +65,34 @@ var pipelinesIndexes = []*col.Index{
 	PipelinesNameIndex,
 }
 
-func ParsePipelineKey(key string) (string, string, error) {
+// ParsePipelineKey expects keys to either be of the form <pipeline>@<id> or
+// <project>/<pipeline>@<id>.
+func ParsePipelineKey(key string) (projectName, pipelineName, id string, err error) {
 	parts := strings.Split(key, "@")
 	if len(parts) != 2 || !uuid.IsUUIDWithoutDashes(parts[1]) {
```

Comment:
right now this doesn't support 3 parts at all

Reply:
I think it does … that’s what  the new-format test case in `TestParsePipelineKey` is for. Am I misreading it?